### ISSUE TITLE
Can't restart Codewind node project in debug mode (using PFE API directly)

### ISSUE THREAD

**Issue by rwalle61** [eclipse/codewind#1968]
**Codewind version:** latest
**OS:** mac
local

following on from https://github.com/eclipse/codewind/issues/1709

**Steps to reproduce:**
1. Create a Node.js project in IDE
2. call `POST /projects/:id/restart` with startMode `debug` on the project
3. see the following error (in PFE logs) and project does not restart
```
[31/01/20 12:32:11 server.js] [DEBUG] POST /api/v1/projects/b3f02540-4425-11ea-86a6-e9bbc9452db4/restart called
[31/01/20 12:32:11 User.js] [DEBUG] Restarting project [object Object] (startMode: debug)
[31/01/20 12:32:42 Turbine] [ERROR] [File Name: /file-watcher/server/dist/controllers/projectsController.js | Function Name: projectAction | Line Number: 972] An error occurred while executing projectAction:
 {"action":"restart","projectID":"b3f02540-4425-11ea-86a6-e9bbc9452db4","startMode":"debug"}
 400: Node.js projects do not support restarting in debug mode.
[31/01/20 12:32:42 /portal/modules/FileWatcher.js] [DEBUG] Received filewatcher module response: { statusCode: 400,
  status: 'failed',
  error:
   { msg: 'Node.js projects do not support restarting in debug mode.' } }
[31/01/20 12:32:42 /portal/routes/projects/restart.route.js] [ERROR] { FilewatcherError: Request error for project b3f02540-4425-11ea-86a6-e9bbc9452db4.
Node.js projects do not support restarting in debug mode.
    at handleProjectActionResponse (/portal/modules/FileWatcher.js:56:727)
    at FileWatcher.performProjectActionRequest (/portal/modules/FileWatcher.js:23:5379)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'FilewatcherError',
  code: 'REQUEST_VALIDATION_ERROR',
  info:
   { name: 'FilewatcherError',
     code: 'REQUEST_VALIDATION_ERROR',
     message:
      'Request error for project b3f02540-4425-11ea-86a6-e9bbc9452db4. Node.js projects do not support restarting in debug mode.' } }
```

Potential fix:
Include `debug` in the Node.js project capabilities: https://github.com/eclipse/codewind/blob/master/src/pfe/file-watcher/server/src/projects/nodejsProject.ts#L26. (Note this unit test will have to be updated too https://github.com/eclipse/codewind/blob/master/src/pfe/file-watcher/server/test/functional-test/configs/project.config.ts#L75)

But is this a genuine problem? The IDEs are currently using `debugNoInit` for Node.js projects. What is the difference between `debug` and `debugNoInit` for Node.js projects?

**Comment by tetchel** [comment]
oh, i misunderstood your comment on the old issue. This is the correct behaviour.

The history of `debug` / `debugNoInit` is that when we first implemented restart-into-debug in Microclimate, we supported Microprofile first. Microprofile servers started in debug mode wait for the debugger to attach **before starting the server**. This means that in Microclimate/Codewind, the project will get stuck `Starting` until the debugger attaches. so, for Microprofile, we can debug the server's initialization (`init`).

When we added support for Spring and Node debug, we wanted to note that the server started before the debugger should attach, the opposite of the Microprofile behaviour. But we'd already used the `debug` start mode for the Microprofile behaviour, so this different behaviour of NOT debugging initialization was named the `debugNoInit` start mode.

Today, Spring supports both start modes, though there's no way to select which one in the IDEs. Node supports only `debugNoInit`, and Microprofile still only `debug`.
